### PR TITLE
Another try for parsing multiple items

### PR DESCRIPTION
**Pull request by hyfc**

### CODE REVIEW

File: docs/tutorial.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ MySpider.run()
 
 Inherit the gain.Spider class. And do some config.
 The "start_url" means the starting point.
-The "frequency" means the concurrent quantity.
+The "concurrency" means the concurrent quantity.
```
Comment (gaojiuli) [inline docs/tutorial.md]: This is useful.

File: gain/selector.py [hunk 2]
```diff
@@ -35,8 +35,12 @@ def parse_detail(self, html):
         d = etree.HTML(html)
         try:
             if self.attr is None:
-                return d.xpath(self.rule)[0].text
-            return d.xpath(self.rule)[0].get(self.attr, None)
+                if len(d.xpath(self.rule)) > 1:
+                    return [entry.text for entry in d.xpath(self.rule)]
+                else:
+                    return d.xpath(self.rule)[0].text
+            return [entry.get(self.attr, None) for entry in d.xpath(self.rule)] if len(d.xpath(self.rule)) > 1 else \
```
Comment (gaojiuli) [inline gain/selector.py]: What's the advantages over the solution of @Jie-OY

Comment (hyfc) [inline gain/selector.py]: This selector organizes all matching items in lists, and leave it to users regarding how to deal with these data.  I think we shouldn't overdo much with scraped data. Keeping it simple is better.

Comment (gaojiuli) [inline gain/selector.py]: > We shouldn't overdo much with scraped data.

I agree with you. But the problem is how to divide items. Please look the conversation [https://github.com/gaojiuli/gain/pull/26#pullrequestreview-45635582](https://github.com/gaojiuli/gain/pull/26#pullrequestreview-45635582)

Maybe you have a better solution for a page contains 10 title and 1 author and a page contains 10 title and 10 author.

**Pull request merged**